### PR TITLE
[improvement](executor)clear unused cgroup path

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Clear unused cgroup path.

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/src/agent/cgroup_cpu_ctl.cpp [hunk 1]
```diff
@@ -157,4 +181,45 @@ Status CgroupV1CpuCtl::add_thread_to_cgroup() {
...
+        }
+    }
+    closedir(query_path_dir);
+
+    // 2 delete unused cgroup path
+    int failed_count = 0;
+    std::string query_path = _cgroup_v1_cpu_query_path.back() != '/'
+                                     ? _cgroup_v1_cpu_query_path + "/"
+                                     : _cgroup_v1_cpu_query_path;
+    for (std::string unused_wg_id : unused_wg_ids) {
```
Comment (HappenLee) [inline be/src/agent/cgroup_cpu_ctl.cpp:210]: `const auto &` to prevent the copy of string.

File: be/src/agent/cgroup_cpu_ctl.cpp [hunk 2]
```diff
@@ -157,4 +181,45 @@ Status CgroupV1CpuCtl::add_thread_to_cgroup() {
     return CgroupCpuCtl::write_cg_sys_file(_cgroup_v1_cpu_tg_task_file, tid, msg, true);
 #endif
 }
+
+Status CgroupV1CpuCtl::delete_unused_cgroup_path(std::set<uint64_t> used_wg_ids) {
```
Comment (HappenLee) [inline be/src/agent/cgroup_cpu_ctl.cpp:185]: `const std::set<uint64_t>& used_wg_ids` better

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**